src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,172 @@
+package duke.util;
+
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.EventCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.ToDoCommand;
+import duke.exceptions.DukeBadDateException;
+import duke.exceptions.DukeBadDateTimeException;
+import duke.exceptions.DukeBadFileException;
+import duke.exceptions.DukeBadIndexException;
+import duke.exceptions.DukeException;
+import duke.exceptions.DukeNoDateException;
+import duke.exceptions.DukeNoDescriptionException;
+import duke.exceptions.DukeNoIndexException;
+import duke.exceptions.DukeNoKeywordException;
+import duke.exceptions.DukeUnknownCommandException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import java.util.Scanner;
+
+public class Parser {
+    protected String command;
+
+    /**
+     * Method to parse a single command.
+     * @param s string to be parsed.
+     * @return parsed command as a Command object.
+     * @throws DukeException if the command is invalid.
+     */
+    public static Command parseCommand(String s) throws DukeException {
+        Scanner sc = new Scanner(s);
+        String w1 = sc.next();
```

Comment:
What do you mean by w1? Try to give more descriptive names so that it will be easy to understand